Sources/Commands/SwiftRunTool.swift

```diff
@@ -137,6 +140,32 @@ public struct SwiftRunTool: SwiftCommand {
                 originalWorkingDirectory: swiftTool.originalWorkingDirectory,
                 arguments: arguments)
 
+        case .debugger:
+            do {
+                let buildSystem = try swiftTool.createBuildSystem(explicitProduct: options.executable)
+                let productName = try findProductName(in: buildSystem.getPackageGraph())
+                if options.shouldBuildTests {
+                    try buildSystem.build(subset: .allIncludingTests)
+                } else if options.shouldBuild {
+                    try buildSystem.build(subset: .product(productName))
+                }
+
+                let executablePath = try swiftTool.buildParameters().buildPath.appending(component: productName)
+
+                // Make sure we are running from the original working directory.
+                let cwd: AbsolutePath? = localFileSystem.currentWorkingDirectory
```

Comment:
What's the reason for the difference of logic here compared with `.run`?  I'm not sure if the net effect will be different, but I wonder if something like being in a different directory and passing `--package-path` to the package would yield a different result here.

Reply:
Oh, I see:  this came from inside `run(
        _ excutablePath: AbsolutePath,
        originalWorkingDirectory: AbsolutePath,
        arguments: [String])`

Reply:
So it does make the logic be the same.